repos/system_upgrade/common/libraries/rpms.py

```diff
@@ -114,3 +115,36 @@ def check_file_modification(config):
     """
     output = _read_rpm_modifications(config)
     return _parse_config_modification(output, config)
+
+
+def get_leapp_packages(source_version=None, component=None, include_deps=False):
+    """
+    Get list of leapp packages.
+
+    :param source_version: a list or string specifying source_versions. If not defined then all packages
+        for current source_target_version will be returned.
+    :param component: a list or string specifying leapp components. If defined then only packages related
+        to the specific component will be returned.
+    :param include_deps: a flag to control whether deps meta packages are to be filtered out.
+    """
+    res = set()
+    # NOTE(ivasilev) mapping has the format component: {source_version: [pkg1, pkg2, ..]}
+    component_map = {'framework': {'7': ['leapp', 'leapp-deps', 'python2-leapp'],
+                                   '8': ['leapp', 'leapp-deps', 'python3-leapp']},
+                     'repository': {'7': ['leapp-upgrade-el7toel8', 'leapp-upgrade-el7toel8-deps'],
+                                    '8': ['leapp-upgrade-el8toel9', 'leapp-upgrade-el8toel9-deps']},
+                     'snactor': {'7': ['snactor'],
+                                 '8': ['snactor']}}
+    source_versions = [source_version] if isinstance(source_version, str) else (source_version or
+                                                                                [get_source_major_version()])
+    components = [component] if isinstance(component, str) else component or list(component_map.keys())
+    for comp in components:
+        for src_version in source_versions:
+            try:
+                res.update(component_map[comp][src_version])
+            except KeyError:
+                continue
+    if not include_deps:
+        # filter out deps packages
+        return sorted([pkg for pkg in res if not pkg.endswith('-deps')])
+    return sorted(res)
```

Comment:
Two things you could do with this code:
```suggestion
    if not include_deps:
        # filter out deps packages
        res = (pkg for pkg in res if not pkg.endswith('-deps'))

    return sorted(res)
```
* [Cosmetic] Generator expression instead of a list comprehension for less cluttered syntax (note this is slower on python2. On python3 it should be the same speed.  It will save memory on both.  But none of this optimization matters for such a short list).

* I'm not always against multiple return statements but this one seems better structured when written this way.  It's a processing of the result before returning.  If a second processing step was added, you would want `res` to pass through both before being returned. These are some general criteria I use for appropriate use of `return`:

  * The common path should exit at the bottom of the function.
  * Exits that can be determined early are fine (examples: cached results, no-ops due to specific parameters).
  * Exits due to errors are fine (some of these are exceptions but others are warnings plus returning from the code). 
  * Other early exits may have their use cases (examples: exiting from a loop, exiting when inside of a large conditional block far from the bottom of the function) but should be looked at on a case by case basis.